lib/pages/trade/bloc/TradePageBloc.dart

```diff
@@ -28,87 +29,54 @@ class TradePageBloc extends Bloc<TradePageEvent, TradePageState> {
     on<SwapTokens>(_mapSwapTokensEventToState);
   }
 
-  get initialState => TradePageState();
-
   void _mapRefreshEventToState(
       PageRefreshEvent event, Emitter<TradePageState> emit) async {
-    emit(state.copyWith(status: Status.loading));
+    emit(state.copyWith(status: BlocStatus.loading));
     try {
       final tokenFromBalance = await walletController
-          .getTokenBalance(state.tokenFrom!.address.value);
+          .getTokenBalance(state.tokenFrom.address.value);
       final tokenToBalance =
-          await walletController.getTokenBalance(state.tokenTo!.address.value);
+          await walletController.getTokenBalance(state.tokenTo.address.value);
       emit(state.copyWith(
           tokenFromBalance: double.parse(tokenFromBalance),
           tokenToBalance: double.parse(tokenToBalance)));
-      final response = await repo.fetchSwapInfo(
-          tokenFrom: state.tokenFrom!.address.value,
-          tokenTo: state.tokenTo!.address.value);
+      final response = await repo.fetchSwaplInfo(
+          tokenFrom: state.tokenFrom.address.value,
+          tokenTo: state.tokenTo.address.value);
       final isSuccess = response.isLeft();
       print("isSuccess - $isSuccess");
       if (isSuccess) {
-        swapController.updateFromAddress(state.tokenFrom!.address.value);
-        swapController.updateToAddress(state.tokenTo!.address.value);
+        swapController.updateFromAddress(state.tokenFrom.address.value);
+        swapController.updateToAddress(state.tokenTo.address.value);
         final swapInfo = response.getLeft().toNullable()!.swapInfo;
-        final transactionInfo =
-            calculateTransactionInfo(swapInfo, state.tokenInputFromAmount);
 
         //do some math
         emit(state.copyWith(
-          status: Status.success,
-          minimumReceived: transactionInfo.minimumReceived!.toDouble(),
-          priceImpact: transactionInfo.priceImpact!.toDouble(),
-          receiveAmount: transactionInfo.receiveAmount!.toDouble(),
-          price: transactionInfo.price!.toDouble(),
-          totalFees: transactionInfo.totalFee!.toDouble(),
+          status: BlocStatus.success,
+          swapInfo: swapInfo,
         ));
       } else {
         final errorMsg = response.getRight().toNullable()!.errorMsg;
         //TODO Create User facing error messages https://athletex.atlassian.net/browse/AX-466
         print(errorMsg);
         emit(state.copyWith(
-          status: Status.error,
-          minimumReceived: 0,
-          priceImpact: 0,
-          receiveAmount: 0,
-          price: 0,
-          totalFees: 0,
+          status: BlocStatus.error,
+          swapInfo: SwapInfo.empty(), 
         ));
       }
     } catch (e) {
-      emit(state.copyWith(status: Status.error));
+      emit(state.copyWith(status: BlocStatus.error, swapInfo: SwapInfo.empty()));
```

Comment:
no need to clear all the info for _every_ error this kind of thing should be more intentional